src/libstd/os/nacl/raw.rs

```diff
@@ -10,160 +10,6 @@
-    use os::unix::raw::{gid_t, uid_t};
-
-    pub type blkcnt_t = i32;
-    pub type blksize_t = i32;
-    pub type ino_t = u32;
-    pub type nlink_t = u32;
-    pub type off_t = i32;
-    pub type time_t = i32;
-
-    #[repr(C)]
-    pub struct stat {
-        pub st_dev: c_ulong,
-        pub st_pad1: [c_long; 3],
-        pub st_ino: ino_t,
-        pub st_mode: mode_t,
-        pub st_nlink: nlink_t,
-        pub st_uid: uid_t,
-        pub st_gid: gid_t,
-        pub st_rdev: c_ulong,
-        pub st_pad2: [c_long; 2],
-        pub st_size: off_t,
-        pub st_pad3: c_long,
-        pub st_atime: time_t,
-        pub st_atime_nsec: c_long,
-        pub st_mtime: time_t,
-        pub st_mtime_nsec: c_long,
-        pub st_ctime: time_t,
-        pub st_ctime_nsec: c_long,
-        pub st_blksize: blksize_t,
-        pub st_blocks: blkcnt_t,
-        pub st_pad5: [c_long; 14],
-    }
-}
-
-#[cfg(target_arch = "aarch64")]
-mod arch {
-    use super::{dev_t, mode_t};
-    use os::raw::{c_long, c_int};
-    use os::unix::raw::{gid_t, uid_t};
-
-    pub type blkcnt_t = i64;
-    pub type blksize_t = i32;
-    pub type ino_t = u64;
-    pub type nlink_t = u32;
-    pub type off_t = i64;
-    pub type time_t = i64;
-
-    #[repr(C)]
-    pub struct stat {
-        pub st_dev: dev_t,
-        pub st_ino: ino_t,
-        pub st_mode: mode_t,
-        pub st_nlink: nlink_t,
-        pub st_uid: uid_t,
-        pub st_gid: gid_t,
-        pub st_rdev: dev_t,
-        pub __pad1: dev_t,
-        pub st_size: off_t,
-        pub st_blksize: blksize_t,
-        pub __pad2: c_int,
-        pub st_blocks: blkcnt_t,
-        pub st_atime: time_t,
-        pub st_atime_nsec: c_long,
-        pub st_mtime: time_t,
-        pub st_mtime_nsec: c_long,
-        pub st_ctime: time_t,
-        pub st_ctime_nsec: c_long,
-        pub __unused: [c_int; 2],
-    }
-}
-
-#[cfg(target_arch = "x86_64")]
-mod arch {
-    use super::{dev_t, mode_t};
-    use os::raw::{c_long, c_int};
-    use os::unix::raw::{gid_t, uid_t};
-
-    pub type blkcnt_t = i64;
-    pub type blksize_t = i64;
-    pub type ino_t = u64;
-    pub type nlink_t = u64;
-    pub type off_t = i64;
-    pub type time_t = i64;
-
-    #[repr(C)]
-    pub struct stat {
-        pub st_dev: dev_t,
-        pub st_ino: ino_t,
-        pub st_nlink: nlink_t,
-        pub st_mode: mode_t,
-        pub st_uid: uid_t,
-        pub st_gid: gid_t,
-        pub __pad0: c_int,
-        pub st_rdev: dev_t,
-        pub st_size: off_t,
-        pub st_blksize: blksize_t,
-        pub st_blocks: blkcnt_t,
-        pub st_atime: time_t,
-        pub st_atime_nsec: c_long,
-        pub st_mtime: time_t,
-        pub st_mtime_nsec: c_long,
-        pub st_ctime: time_t,
-        pub st_ctime_nsec: c_long,
-        pub __unused: [c_long; 3],
-    }
-}
+// Reuse the definitions from libc.
+pub use libc::{dev_t, ino_t, mode_t, nlink_t, uid_t, gid_t, off_t, time_t,
+               blkcnt_t, blksize_t, stat};
```

Comment:
Due to the way that these are defined in terms of where the origin is it'll be much better for these definitions to stay here instead of inlining the liblibc version.